### ISSUE TITLE
[Bug][Light Node]: No chain head errors during sync with trusted peer and hash

### ISSUE THREAD

**Issue by Bidon15** [celestiaorg/celestia-node#510]
### Celestia Node version

0806a0a1080ee980a5f645494fd38d6f274f91e3

### OS

macOS 12.0.1

### Install tools

make install

### Steps to reproduce it

1. Make install
2. Initialize the node with a trusted peer and hash
- `/ip4/46.101.22.123/tcp/2121/p2p/12D3KooWJFmUfzqXTXAQwhAhcXfwie2HUcrtNj1KNgLjbprZuwHR`
- `4632277C441CA6155C4374AC56048CF4CFE3CBB2476E07A548644435980D5E17`
4. Start the node

### Expected result

Node is syncing from genesis block

### Actual result

Node is throwing errors like `no chain head` during sync

### Relevant log output

```shell
2022-03-10T13:50:14.323+0200	ERROR	node/services	services/service.go:112	initializing store failed: protocol not supported
2022-03-10T13:50:14.323+0200	INFO	watchdog	go-watchdog@v1.2.0/watermarks.go:16	initialized watermark watchdog policy; watermarks: []; thresholds: [8589934592 10307921510 12025908428 14602888806 15461882265 15891378995 16320875724]
2022-03-10T13:50:14.323+0200	INFO	watchdog	go-watchdog@v1.2.0/watchdog.go:471	initialized heap profile capture; threshold: 0.900000; max captures: 10; dir: /Users/bidon4/.celestia-light
2022-03-10T13:50:14.325+0200	ERROR	header	header/sync.go:170	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:50:14.325+0200	INFO	header	header/p2p_server.go:37	p2p-server: listening for inbound header requests
2022-03-10T13:50:14.326+0200	INFO	node	node/node.go:109	started Light Node
The p2p host is listening on:
*  /ip4/192.168.0.125/tcp/2121/p2p/12D3KooWKyHiaiTzixwyyxeYcUbGt4RAPfv5mvW738wfhJmUyPzQ
*  /ip6/::1/tcp/2121/p2p/12D3KooWKyHiaiTzixwyyxeYcUbGt4RAPfv5mvW738wfhJmUyPzQ
2022-03-10T13:50:16.732+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604005, "hash": "21D9DCE80D5B9E3F49116A015CCAB91F42C2659AAC14F7BA3A67725D1676B61D", "err": "header/store: no chain head"}
2022-03-10T13:50:16.732+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:50:22.471+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604006, "hash": "6C2DA93739405AFB454FF9FD37F41E034B4074550963C3B31EF6DAA6DA3F4988", "err": "header/store: no chain head"}
2022-03-10T13:50:22.471+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:50:38.467+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604007, "hash": "A97CD260C34A5A3876249473C78CF854F2BF1C85B172DC6703ECD62389C6F0D1", "err": "header/store: no chain head"}
2022-03-10T13:50:38.467+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:50:54.477+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604008, "hash": "F93D9397FB4756280B9EFC74ECA4CCFC79CC4E799322874335ED0E82D25C974D", "err": "header/store: no chain head"}
2022-03-10T13:50:54.477+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:51:10.248+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604009, "hash": "71B00682F16FCE39E01774F7522629706DB59D77C90B85FBD5A8EF147D0E45A6", "err": "header/store: no chain head"}
2022-03-10T13:51:10.248+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
2022-03-10T13:51:16.600+0200	ERROR	header	header/sync.go:193	appending header	{"height": 604010, "hash": "90A5E19788833C2C97D299FF01DD3E359185B5BE841FD93F8E7EED57256B474E", "err": "header/store: no chain head"}
2022-03-10T13:51:16.600+0200	ERROR	header	header/sync.go:203	getting trusted head	{"err": "header/store: no chain head"}
```

### Notes

This is also reproducible in v0.2.0

Logs for main branch + peer + hash(genesis): https://gist.github.com/Bidon15/6e354d1a38c38136e686d9113f8ff397
Logs for v0.2.0 + peer + hash(genesis): https://gist.github.com/Bidon15/379a00a98dd0d3e819f221bab9b6e651

**Comment by Bidon15** [comment]
if initializing without any trusted peers / hash - sync is working as expected (errors might occur due to inactive nyc bootstrapper, which is a know issue that is being resolved rn)

**Comment by dieuvu01** [comment]
Please check the log:
root@vmi829755:~# sudo journalctl -u celestia-lightd.service -f
-- Logs begin at Wed 2022-03-23 01:20:31 CDT. --
Mar 23 05:57:26 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:26                                                                                                             .215-0500        ERROR        header        header/sync.go:193        appending                                                                                                              header        {"height": 687186, "hash": "1F8D1E6FD157ED85BDC7A2DFBD799EE98FC580                                                                                                             7CEF58315287C92AC538CF1F52", "err": "header/store: no chain head"}
Mar 23 05:57:26 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:26                                                                                                             .216-0500        ERROR        header        header/sync.go:203        getting tr                                                                                                             usted head        {"err": "header/store: no chain head"}
Mar 23 05:57:32 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:32                                                                                                             .312-0500        ERROR        header        header/sync.go:193        appending                                                                                                              header        {"height": 687187, "hash": "1D0FE2ED35878C0591B6343AB1D786EC87ADAA                                                                                                             C7219ED48A1CB112D0DC074B8C", "err": "header/store: no chain head"}
Mar 23 05:57:32 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:32                                                                                                             .312-0500        ERROR        header        header/sync.go:203        getting tr                                                                                                             usted head        {"err": "header/store: no chain head"}
Mar 23 05:57:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:48                                                                                                             .353-0500        ERROR        header        header/sync.go:193        appending                                                                                                              header        {"height": 687188, "hash": "E23DD5B3B29943D2CE5CF00FCB8C5A53249C02                                                                                                             2A1588A7F2111BB588ED838D9A", "err": "header/store: no chain head"}
Mar 23 05:57:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:48                                                                                                             .353-0500        ERROR        header        header/sync.go:203        getting tr                                                                                                             usted head        {"err": "header/store: no chain head"}
Mar 23 05:57:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:54                                                                                                             .240-0500        ERROR        header        header/sync.go:193        appending                                                                                                              header        {"height": 687189, "hash": "8531581891E7B187E2FA94C0C380096CAB4F77                                                                                                             446BE1DEEAE8CB9560446C3BCC", "err": "header/store: no chain head"}
Mar 23 05:57:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:57:54                                                                                                             .241-0500        ERROR        header        header/sync.go:203        getting tr                                                                                                             usted head        {"err": "header/store: no chain head"}
Mar 23 05:58:00 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:58:00                                                                                                             .548-0500        ERROR        header        header/sync.go:193        appending                                                                                                              header        {"height": 687190, "hash": "7697EB6081BD29557DAE4E530AF2D4F8411EB4                                                                                                             67B9D1A0B7B8390603310C427C", "err": "header/store: no chain head"}
Mar 23 05:58:00 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T05:58:00                                                                                                             .548-0500        ERROR        header        header/sync.go:203        getting tr                                                                                                             usted head        {"err": "header/store: no chain head"}

**Comment by dieuvu01** [comment]
Other log:

root@vmi829755:~# sudo journalctl -u celestia-lightd.service -n 100
-- Logs begin at Wed 2022-03-23 01:20:31 CDT, end at Wed 2022-03-23 06:17:17 CD>
Mar 23 06:16:43 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:43 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:44 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:44 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:46 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:46 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:47 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:47 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:49 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:49 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:4>
Mar 23 06:16:50 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
Mar 23 06:16:50 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:5>
lines 1-23...skipping...
-- Logs begin at Wed 2022-03-23 01:20:31 CDT, end at Wed 2022-03-23 06:17:17 CDT. --
Mar 23 06:16:43 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:43.906-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:43 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:43.907-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:44 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:44.468-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:44 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:44.468-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:45.091-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:45.092-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:45.897-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:45 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:45.897-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:46 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:46.438-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:46 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:46.439-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:47 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:47.214-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:47 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:47.215-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:48.114-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:48 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:48.114-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:49 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:49.215-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:49 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:49.216-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:50 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:50.128-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:50 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:50.128-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.221-0500        ERROR        header        header/sync.go:193        appending header        {"height": 687>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.226-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.344-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.344-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.509-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:51 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:51.509-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:52 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:52.460-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:52 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:52.460-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:52 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:52.828-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:52 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:52.828-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:53 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:53.405-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:53 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:53.405-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:54.098-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:54.099-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:54.581-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:54 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:54.581-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:55 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:55.157-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:55 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:55.157-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:56 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:56.357-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:56 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:56.357-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:56 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:56.806-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:56 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:56.806-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:57 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:57.577-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:57 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:57.577-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:58 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:58.102-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:58 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:58.103-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:58 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:58.750-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
Mar 23 06:16:58 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:58.750-0500        ERROR        header        header/sync.go:203        getting trusted head        {"err": "h>
Mar 23 06:16:59 vmi829755.contaboserver.net celestia[21145]: 2022-03-23T06:16:59.476-0500        ERROR        header        header/sync.go:193        appending header        {"height": 659>
lines 1-48

**Comment by renaynay** [comment]
Resolved by #539
